main.sh

```diff
@@ -342,9 +344,18 @@ if [[ "${INPUT_TAR/all/${platform}}" == "${platform}" ]] || [[ "${INPUT_ZIP/all/
     cwd=$(pwd)
     tmpdir=$(mktemp -d)
     mkdir "${tmpdir:?}/${archive}"
-    filenames=("${bins[@]}")
+    if [[ -n "${bin_leading_dir}" ]]; then
+        filenames=("${bin_leading_dir%%/*}")
+    else
+        filenames=("${bins[@]}")
+    fi
     for bin_exe in "${bins[@]}"; do
-        cp "${target_dir}/${bin_exe}" "${tmpdir}/${archive}"/
+        if [[ -n "${bin_leading_dir}" ]]; then
+            x mkdir -p "${tmpdir}/${archive}/${bin_leading_dir}"/
```

Comment:
This `mkdir` could also be moved before the loop.

Reply:
Update: move `mkdir` line out of `for bin_exe` loop, move to before the for loop.